File: src/main/java/dev/wirlie/spigot/glist/hooks/AbstractHook.java
package dev.wirlie.spigot.glist.hooks;

import dev.wirlie.spigot.glist.EnhancedBCLBridge;
import org.bukkit.entity.Player;

public interface AbstractHook {

    void sendStateToBridge(Player player, StateNotificationType type, boolean newValue);

    void sendAllPlayersStateToBridge();

    void sendPlayerToBridge(Player player);

    void registerListeners(EnhancedBCLBridge bridge);

}

File: src/main/kotlin/dev/wirlie/spigot/glist/hooks/AbstractHook.kt
package dev.wirlie.spigot.glist.hooks

import dev.wirlie.spigot.glist.EnhancedBCLBridge
import org.bukkit.entity.Player

interface AbstractHook {

    fun sendStateToBridge(player: Player, type: StateNotificationType, newValue: Boolean)

    fun sendAllPlayersStateToBridge()

    fun sendPlayerToBridge(player: Player)

    fun registerListeners(bridge: EnhancedBCLBridge)

}
